rocketchat-desktop: build from source, update

Mynacol opened the pull request:
This switches to building Rocketchat from source. If wanted, we can move the previous package to rocketchat-desktop-bin.
As such, I also take over the maintainership of this package.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [ ] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Mynacol committed:
rocketchat-desktop: build from source, update

Mynacol commented:
The build is somehow broken on macos. The initial error was missing libtool. But even if libtool is provided, the build complains about unknown cmdline flag `-static`...

Disabling macos for now.

Mynacol commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 469399`
Commit: [`9100d470f68822b723975fa28a84cd1dc9a28745`](https://github.com/NixOS/nixpkgs/commit/9100d470f68822b723975fa28a84cd1dc9a28745) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/9100d470f68822b723975fa28a84cd1dc9a28745..pull/469399/head))
Merge: [`7e465c3b02248c9cd0b684597da479bf47c20239`](https://github.com/NixOS/nixpkgs/commit/7e465c3b02248c9cd0b684597da479bf47c20239)

Logs: https://github.com/Mynacol/nixpkgs-review-gha/actions/runs/20082017066


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>rocketchat-desktop</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>rocketchat-desktop</li>
  </ul>
</details>

SuperSandro2000 commented on pkgs/by-name/ro/rocketchat-desktop/package.nix:
```diff
@@ -1,104 +1,148 @@
+
+    for icon in build/icons/*.png
+    do
+      install -Dm644 $icon $out/share/icons/hicolor/$(basename ''${icon%.png})/apps/${pname}.png
```
```suggestion
      install -Dm644 $icon $out/share/icons/hicolor/$(basename ''${icon%.png})/apps/rocketchat-desktop.png
```

SuperSandro2000 committed:
rocketchat-desktop: avoid substituting pname/meta.mainProgram

Doing this for pname is an anti pattern and meta.mainProgram could be done with finalAttrs, but lets keep things simple.

SuperSandro2000 reviewed (approved):
We support one more platform than before and darwin was not supported before, so to my current knowledge, I see no reason to keep the bin variant.
